sale_order_revision/model/sale_order.py

```diff
@@ -64,6 +69,17 @@ def copy_quotation(self):
         msg = _('New revision created: %s') % self.name
         self.message_post(body=msg)
         old_revision.message_post(body=msg)
+
+        # set stored procurement group id on old order
+        old_revision.write({'procurement_group_id': procurement_group_id})
```

Comment:
I don't think we should switch procurement group. It's good to have the procurement group set on the newest revision, not in the old one.